pulsar_spectra/spectral_fit.py

```diff
@@ -102,10 +103,83 @@ def plot_fit(freqs_MHz, fluxs_mJy, flux_errs_mJy, ref, model, iminuit_result, fi
+    fit_info : `str`
+        The string to label the fit with from :py:meth:`pulsar_spectra.spectral_fit.iminuit_fit_spectral_model`.
+    plot_error : `boolean`, optional
+        If you want to include the fit error in the plot. |br| Default: True.
+    save_name : `str`, optional
+        The name of the saved plot. |br| Default: "fit.png".
+    """
+    # Set up plot
+    plotsize = 3.2
+    fig, ax = plt.subplots(figsize=(plotsize*4/3, plotsize))
+    marker_scale = 0.7
+    capsize = 1.5
+    errorbar_linewidth = 0.7
+    marker_border_thickness = 0.5
+    custom_cycler = (cycler(color = ["#006ddb", "#24ff24",'r',"#920000","#6db6ff","#ff6db6",'m',"#b6dbff","#009292","#b66dff","#db6d00", 'c',"#ffb6db","#004949",'k'])
+                    + cycler(marker = [            'o', '^', 'D', 's', 'p', '*', 'v', 'd', 'P',  'h', '>', 'H', 'X', '<', 'x'])
+                    + cycler(markersize = np.array([6,   7,   5,   5.5, 6.5, 9,   7,   7,   7.5,  7,   7,   7,   7.5,   7,   7])*marker_scale))
+    ax.set_prop_cycle(custom_cycler)
+
+    # Add data
+    data_dict = convert_cat_list_to_dict({"dummy_pulsar":[freqs_MHz, fluxs_mJy, flux_errs_mJy, ref]})["dummy_pulsar"]
+    for ref in data_dict.keys():
+        freqs_ref = np.array(data_dict[ref]['Frequency MHz'])
+        fluxs_ref = np.array(data_dict[ref]['Flux Density mJy'])
+        flux_errs_ref = np.array(data_dict[ref]['Flux Density error mJy'])
+        (_, caps, _) = ax.errorbar(freqs_ref, fluxs_ref, yerr=flux_errs_ref, linestyle='None', mec='k', markeredgewidth=marker_border_thickness, elinewidth=errorbar_linewidth, capsize=capsize, label=ref.replace('_',' '))
+        for cap in caps:
+            cap.set_markeredgewidth(errorbar_linewidth)
+
+    # Create fit line
+    fitted_freq = np.logspace(np.log10(min(freqs_MHz)), np.log10(max(freqs_MHz)), 100)
+    if iminuit_result.valid:
+        fitted_flux, fitted_flux_cov = propagate(lambda p: model(fitted_freq * 1e6, *p) * 1e3, iminuit_result.values, iminuit_result.covariance)
+    else:
+        # No convariance values so use old method
+        fitted_flux = model(fitted_freq * 1e6, *iminuit_result.values) * 1e3
+    # Plot fit line
+    ax.plot(fitted_freq, fitted_flux, 'k--', label=fit_info.split()[0].replace('_', ' '))
+    if plot_error and iminuit_result.valid:
+        # draw 1 sigma error band
+        fitted_flux_prop = np.diag(fitted_flux_cov) ** 0.5
+        ax.fill_between(fitted_freq, fitted_flux - fitted_flux_prop, fitted_flux + fitted_flux_prop, facecolor="C1", alpha=0.5)
+
+    # Format plot and save
+    plt.xscale('log')
+    plt.yscale('log')
+    ax.get_xaxis().set_major_formatter(ScalarFormatter())
+    ax.get_yaxis().set_major_formatter(ScalarFormatter())
+    ax.tick_params(which='both', direction='in', top=1, right=1)
+    plt.xlabel('Frequency (MHz)')
+    plt.ylabel('Flux Density (mJy)')
+    plt.legend(loc='lower left', ncol=2, fontsize=6)
```

Comment:
I don't think you need a whole new function because this is the only change right?

Reply:
The plot 4x5 and slightly higher resolution can probably be done for both types